mwtext/content_transformers/wikidata2words.py

```diff
@@ -31,13 +31,21 @@ def transform(self, content):
         doc = json.loads(content)
         entity = mwbase.Entity.from_json(doc)
         claims_tuples = list(self._extract_property_values(entity))
+        if self.instanceof_nonzero_namespace(claims_tuples):
+            return None
         claims_tuples.sort(key=self.get_claim_pid_index)
         return list(chain(*claims_tuples))
 
     def get_claim_pid_index(self, claims_tuple):
         pid = claims_tuple[0]
         return self.pid_order_map.get(pid, len(self.pid_order_map))
 
+    def instanceof_nonzero_namespace(self, claims_tuples):
+        namespace_QIDs =  ['Q4167836', 'Q11266439', 'Q4663903']
+        return any((claims_tuple[0] == 'P31' and 
+                    claims_tuple[1] in namespace_QIDs)
+                   for claims_tuple in claims_tuples)
+
```

Comment:
What's going on here?  It seems the function is checking for a namespace, but then it is matching against particular Qids?

Reply:
most of the Wikidata items that represent the pages with namespaces we are not interested in are "instance of" those QIDs. 
Another option I discussed with Isaac was to check all the wikis in the sitelink to see if each of them had a prefix of form "<namespace name>:".

Reply:
in the past, I've had success with filtering any instances or subclasses of this https://www.wikidata.org/wiki/Q17442446

Reply:
but in order to do that you have to have a list of all the subclasses .... the wikidata query service typically times out on a query like that .... but maybe you have some other means.  I parse all the subclass edges out of a wikidata dump as a preprocessing step (every statement that has P279), but its not super fast.

Reply:
@galtay you mean filtering out any item that is `instance of` or `subclass of` Q17442446 and it's subclasses?

Reply:
yes, that's the idea.  i'm not sure exactly what your filtering criteria are, but its one way to do these sorts of things.  you can see the subclasses in graph form here --  https://angryloki.github.io/wikidata-graph-builder/?property=P279&item=Q17442446&mode=reverse

Reply:
I had a quick chat with @geohci about this.  I think that the right approach is to write separate script that gathers all of the relevant Qids and puts them into a data file and then we can load that data file as part of the filter method.  

This is a similar pattern to the one we used for extracting idioms for use in revscoring.  See https://github.com/wikimedia/revscoring/blob/master/revscoring/utilities/fetch_idioms.py (this script gathers idioms from wiktionary), https://github.com/wikimedia/revscoring/blob/master/revscoring/assets/enwiktionary_idioms.txt (here is the data file with the idioms), and https://github.com/wikimedia/revscoring/blob/master/revscoring/languages/english.py#L278 (here is where the data file is loaded and used).